src/Compilers/Core/CodeAnalysisTest/AsyncQueueTests.cs

```diff
@@ -170,41 +170,31 @@ public async Task DequeueAsyncWithCancellation()
         }
 
         [Fact]
-        public async Task DequeueAsyncWithCancellationAfterComplete()
+        public async Task EnqueueAfterDequeueAsyncWithCancellation()
         {
             var queue = new AsyncQueue<int>();
             var cts = new CancellationTokenSource();
             var task = queue.DequeueAsync(cts.Token);
-            Assert.False(task.IsCompleted);
-            queue.Enqueue(42);
-            await task.ConfigureAwait(false);
+            Assert.False(task.IsCanceled);
             cts.Cancel();
+            await Assert.ThrowsAsync<TaskCanceledException>(() => task);
+            Assert.Equal(TaskStatus.Canceled, task.Status);
+
+            queue.Enqueue(1);
+            Assert.True(queue.TryDequeue(out var value));
+            Assert.Equal(1, value);
         }
 
         [Fact]
-        public async Task TaskCompletesAsyncWithComplete()
+        public async Task DequeueAsyncWithCancellationAfterComplete()
         {
             var queue = new AsyncQueue<int>();
-
-            var tcs = new TaskCompletionSource<bool>();
-            var task = queue.DequeueAsync().ContinueWith(
-                t =>
-                {
-                    tcs.Task.Wait();
-                    return 0;
-                },
-                default(CancellationToken),
-                TaskContinuationOptions.ExecuteSynchronously,
-                TaskScheduler.Default);
-
-            queue.Complete();
-            Assert.False(queue.WhenCompletedTask.IsCompleted);
-            tcs.SetResult(true);
-            await queue.WhenCompletedTask.ConfigureAwait(false);
-
-            // The AsyncQueue<T>.Task property won't complete until all of the 
-            // existing DequeueAsync values have also completed.
-            Assert.True(task.IsCompleted);
+            var cts = new CancellationTokenSource();
+            var task = queue.DequeueAsync(cts.Token);
+            Assert.False(task.IsCompleted);
+            queue.Enqueue(42);
+            await task.ConfigureAwait(false);
+            cts.Cancel();
```

Comment:
Bit pedantic but can we add a final assert that the task ran to completion?

Reply:
😕 If I saw such an assertion when reading through the code, I would be concerned that the author of the test intended to cover some other scenario and mistakenly wrote a trivially-true assertion (the `await` would never complete if `task` was not complete).

Reply:
➡️ No action taken